packages/agents/src/functions/DeveloperPlanner.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  Agent,
+  AgentFunctionResult,
+  LlmApi,
+  ChatLogs,
+  Tokenizer,
+  ChatMessageBuilder,
+} from "@evo-ninja/agent-utils";
+import { AgentFunctionBase } from "../AgentFunctionBase";
+import { AgentBaseContext } from "../AgentBase";
+
+interface FunctionParams {
+  goal: string;
+  context: string;
+}
+
+export class DeveloperPlanner extends AgentFunctionBase<FunctionParams> {
+  constructor(
+    private _llm: LlmApi,
+    private _tokenizer: Tokenizer
+  ) {
+    super();
+  }
+
+  get name() {
+    return "developerPlanner";
```

Comment:
Names of functions should be verbs and not nouns. Something like `planDevelopmentTasks` could be a better name